cpu/kinetis_common/include/periph_cpu.h

```diff
@@ -309,10 +309,18 @@ enum {
 /** @} */
 
 /**
+ * @brief UART hardware module types
+ */
+enum {
+    KINETIS_UART,   /**< Kinetis UART module type */
+    KINETIS_LPUART, /**< Kinetis Low-power UART (LPUART) module type */
+};
+
+/**
  * @brief UART module configuration options
  */
 typedef struct {
-    UART_Type *dev;               /**< Pointer to module hardware registers */
+    void *dev;                    /**< Pointer to module hardware registers */
```

Comment:
That's interesting. I'm wondering if the same strategy could also be applied to STM32. Some macros make the driver implementation harder to read/understand btw.